code/segmentedtickerplant/datastripe.q

```diff
@@ -22,3 +22,29 @@ initdatastripe:{
      };
 
 if[.ds.datastripe;.proc.addinitlist[(`initdatastripe;`)]];
+
+\d .stpps
+//function to grab where clause from config tables
+segmentfilterfunc:{[tbl;segid] 
+  string first .stpps.segmentfiltermap[exec wcRef from .stpps.segmentconfig where table=tbl , segmentID=segid]};
+// Function to use segmentfilterfunc and return "" incase of error
+segmentfilter:{[tbl;segid]
+  output:.[segmentfilterfunc;
+  (tbl;segid) ;
+  "" ];
+  $[count output>0;output;""]};
+
+  // Function to subscribe to particular segment using segmentID based on .u.sub
+subsegment:{[tbl;segid];
+//tablename and segmentid used to get filters
+//filteroutput needs to be tested
+  if[tbl~`;:.z.s[;segid] each .stpps.t];
+   if[not tbl in .stpps.t;
+     .lg.e[`sub;m:"Table ",string[tbl]," not in list of stp pub/sub tables"];
+     :(tbl;m)
+  ];
+  $[segid~`;[filter:`]; [filter: segmentfilter[tbl;segid]]];
```

Comment:
segid is not a symbol, its an int. better to use `null segid` for this conditional. Its more general.

Reply:
```
  filter:$[null segid; `;segmentfilter[tbl;segid]];
  [filter~`;.stpps.suball[tbl]; .ps.subtablefiltered[string[tbl];filter;""]]`
```
Made the changes requested but when you use .stpps.subsegment[`quote;`] it gives a type error. `segid ~ `  would allow the function to act .u.sub[`quote;`] where you sub to quote with no filter. At the moment the only way I think you can sub to a table without a filter is to trigger the error trap in `.stpps.segmentfilter[quote;quote]`.